Formula/kim-api.rb

```diff
@@ -30,23 +30,33 @@ def install
     inreplace "cmake/items-macros.cmake.in" do |s|
       s.gsub!(/file\(COPY ([^ ]+) DESTINATION ([^ ]*)\)/, 'configure_file(\1 \2 COPYONLY)')
     end
-    args = std_cmake_args
+    args = std_cmake_args + [
+      # adjust libexec dir
+      "-DCMAKE_INSTALL_LIBEXECDIR=lib",
+      # adjust directories for system collection
+      "-DKIM_API_SYSTEM_MODEL_DRIVERS_DIR=:#{HOMEBREW_PREFIX}/lib/openkim-models/model-drivers",
+      "-DKIM_API_SYSTEM_PORTABLE_MODELS_DIR=:#{HOMEBREW_PREFIX}/lib/openkim-models/portable-models",
+      "-DKIM_API_SYSTEM_SIMULATOR_MODELS_DIR=:#{HOMEBREW_PREFIX}/lib/openkim-models/simulator-models",
+      # adjust zsh completion install
+      "-DZSH_COMPLETION_COMPLETIONSDIR=#{zsh_completion}",
+      "-DBASH_COMPLETION_COMPLETIONSDIR=#{bash_completion}",
+    ]
     # adjust compiler settings for package
-    args << "-DKIM_API_CMAKE_C_COMPILER=/usr/bin/clang"
-    args << "-DKIM_API_CMAKE_CXX_COMPILER=/usr/bin/clang++"
-    # adjust libexec dir
-    args << "-DCMAKE_INSTALL_LIBEXECDIR=lib"
-    # adjust directories for system collection
-    args << "-DKIM_API_SYSTEM_MODEL_DRIVERS_DIR=:#{HOMEBREW_PREFIX}/lib/openkim-models/model-drivers"
-    args << "-DKIM_API_SYSTEM_PORTABLE_MODELS_DIR=:#{HOMEBREW_PREFIX}/lib/openkim-models/portable-models"
-    args << "-DKIM_API_SYSTEM_SIMULATOR_MODELS_DIR=:#{HOMEBREW_PREFIX}/lib/openkim-models/simulator-models"
-    # adjust zsh completion install
-    args << "-DZSH_COMPLETION_COMPLETIONSDIR=#{zsh_completion}"
-    args << "-DBASH_COMPLETION_COMPLETIONSDIR=#{bash_completion}"
-    system "cmake", ".", *args
-    system "make"
-    system "make", "docs"
-    system "make", "install"
+    on_macos do
+      args << "-DKIM_API_CMAKE_C_COMPILER=/usr/bin/clang"
```

Comment:
Can we try to use `ENV.cc`/`ENV.cxx`?  Does that cause an issue with compiler shims?

Reply:
I haven't tried. I just reused most of logic that was already part of formula as I didn't know exact reason why some CMake args were used. May need to check commit history to see if there was an explanation somewhere.

Reply:
Didn't work as the input is interpreted as a path, so it tries to find relative to build directory:
```
    /tmp/kim-api-20210808-9530-16wphy9/kim-api-2.2.1/build/clang

  is not a full path to an existing compiler tool.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.
```

Reply:
Is it possible for us to give it the full shim path instead of `ENV.cc`?  That would still let us use the shim, even though if it's a bit more verbose.

Reply:
That would probably cause failure on detected shim path as the value seems to be saved somewhere in `kim-api` since it is reused when building `openkim-models`.

Reply:
Do we know if the file(s) the shim paths are in are binaries?  If so then I guess we don't have much choice.

Reply:
As far as I can tell, this variable is not used during build time, but is specifically set to be saved inside the final output.

For example, if I run `brew install kim-api -s --cc gcc-11`, 02.make.cc log shows that shim g++-11 is used:
```
g++-11 called with: CMakeCXXCompilerId.cpp
superenv added:    -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk --sysroot=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -isystem/usr/local/include -isystem/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers -L/usr/local/opt/gcc/lib/gcc/11 -L/usr/local/lib -L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries -Wl,-headerpad_max_install_names
superenv executed: g++-11 CMakeCXXCompilerId.cpp -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk --sysroot=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -isystem/usr/local/include -isystem/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers -L/usr/local/opt/gcc/lib/gcc/11 -L/usr/local/lib -L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries -Wl,-headerpad_max_install_names
```

So, using the existing CMake vars to set it should be similar to inreplacing the shim paths in final output. However, the CMake vars are cleaner to use directly.

From INSTALL file:
```
* KIM_API_CMAKE_C_COMPILER (="${CMAKE_C_COMPILER}") This value is recorded and
  used after installation for compilation of items (MDs, PMs, SMs, etc.) to
  ensure binary compatibility with the installed KIM API.

* KIM_API_CMAKE_CXX_COMPILER (="${CMAKE_CXX_COMPILER}") This value is recorded
  and used after installation for compilation of items (MDs, PMs, SMs, etc.)
  to ensure binary compatibility with the installed KIM API.
```